File: ArrayList/app/src/main/java/com/bcil/arraylist/MainActivity.java
package com.bcil.arraylist;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.LinearLayout;

import com.bcil.arraylist.listener.AnimalListener;
import com.bcil.arraylist.listener.ContactListener;
import com.bcil.arraylist.listener.NumberListener;

public class MainActivity extends AppCompatActivity {
   LinearLayout llinerContact;
   LinearLayout llinerNumber;
   LinearLayout llinerAnimal;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        llinerNumber = (LinearLayout)findViewById(R.id.llinerNumber);
        llinerContact  = (LinearLayout)findViewById(R.id.llinerContact);
        llinerAnimal  = (LinearLayout)findViewById(R.id.llinerAnimal);
        llinerNumber.setOnClickListener(new NumberListener(this));
        llinerContact.setOnClickListener(new ContactListener(this));
        llinerAnimal.setOnClickListener(new AnimalListener(this));
    }
}

File: MVPDemo/app/src/main/java/com/melayer/demo/mvpdemo/login/ui/MainActivity.kt
package com.melayer.demo.mvpdemo.login.ui

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import com.melayer.demo.mvpdemo.R
import com.melayer.demo.mvpdemo.login.domain.Login
import com.melayer.demo.mvpdemo.login.mvp.LoginContractor
import kotlinx.android.synthetic.main.activity_main.*
import org.koin.android.ext.android.inject

class MainActivity : AppCompatActivity(),LoginContractor.View {
    override val presenter: LoginContractor.Presenter by inject()

    override fun doLogin(name: String) {
        text.text="Welcome "+name
    }

    override fun errorLogin() {
        Toast.makeText(applicationContext,"Try again",Toast.LENGTH_LONG)
                .show()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        presenter.getLogin(getData())
    }
    override fun onResume() {
        presenter.view = this
        super.onResume()
    }
    private fun getData(): Login {
        val login=Login()
        login.emailId="saindane.rajan@gmail.com"
        login.password="rajan123"
        return login
    }
}
